logr.go

```diff
@@ -33,16 +34,16 @@ type Logger interface {
 	// example, logger.Info() produces the same result as logger.V(0).Info.
 	InfoLogger
 
-	// Error logs a error message.  This is behaviorally akin to fmt.Print.
-	Error(args ...interface{})
-
-	// Errorf logs a formatted error message.
-	Errorf(format string, args ...interface{})
+	// Error logs an error, with the given message and key/value pairs as context.
+	Error(err error, msg string, keysAndValues ...interface{})
 
 	// V returns an InfoLogger value for a specific verbosity level.  A higher
 	// verbosity level means a log message is less important.
 	V(level int) InfoLogger
 
-	// NewWithPrefix returns a Logger which prefixes all messages.
-	NewWithPrefix(prefix string) Logger
+	// WithTags adds some key-value pairs of context to a logger.
+	WithTags(keysAndValues ...interface{}) Logger
+
+	// WithName adds a new prefix to the logger's name.
```

Comment:
We should document the implied results.  E.g. is it a strict prefix (if I want a space I need to add it myself) or is it tracked as a "list" of some sort such that the result is delimited?  And if so, is the delimiting specified?  Are there characters I can't use in the `WithName` ?

e.g. if I call `logger.WithName("foo").WithName("bar").Info("message")` - what can I assume about the result?